lib/resultuploaders/result_uploader.rb

```diff
@@ -10,24 +11,48 @@ class ResultUploader
     # UploaderFactory
     #
     class UploaderFactory
-      UPLOADERS = {
-        dropbox: Dropbox
-      }.freeze
+      def self.initialize(logger)
+        @uploades = {}
+
+        Dir['./lib/resultuploaders/*.rb'].sort.each do |file|
```

Comment:
Strictly I don't think it's assured that the current working directory is at the project root.